src/WebJobs.Script.WebHost/Metrics/LinuxContainerMetricsPublisher.cs

```diff
@@ -0,0 +1,227 @@
+
+        internal void OnFunctionMetricsPublishTimer(object state)
+        {
+            PublishActivity(_currentMemoryActivities, _memoryActivities, PublishMemoryActivityPath);
+            PublishActivity(_currentFunctionActivities, _functionActivities, PublishFunctionActivityPath);
+        }
+
+        private void PublishActivity<T>(ConcurrentQueue<T> currentActivities, BlockingCollection<T> activitiesToProcess, string publishPath)
+        {
+            try
+            {
+                if (currentActivities.Any())
+                {
+                    SendRequest(currentActivities, publishPath);
+                }
+                DrainActivities(currentActivities, activitiesToProcess);
+            }
+            catch (Exception ex)
+            {
+                _errorCount++;
+                _logTraceEvent(LogLevel.Warning, string.Format("LinuxContainerMetricsPublisher : Error publishing activites. Exception : {0}", ex.ToString()));
+                if (_errorCount < _maxErrorLimit)
+                {
+                    _logTraceEvent(LogLevel.Warning, string.Format("LinuxContainerMetricsPublisher : Error count {0} within threshold. will retry later.", _errorCount));
+                }
+                else
+                {
+                    _logTraceEvent(LogLevel.Error, string.Format("LinuxContainerMetricsPublisher : Max error limit reached. Draining activities for {0}", _containerName));
+                    DrainActivities(currentActivities, activitiesToProcess);
+                    _errorCount = 0;
+                }
+            }
+        }
+
+        private void SendRequest<T>(ConcurrentQueue<T> activitiesToPublish, string publishPath)
+        {
+            var request = BuildRequest(HttpMethod.Post, publishPath, activitiesToPublish.ToArray());
+            HttpResponseMessage response = _httpClient.SendAsync(request).GetAwaiter().GetResult();
+            response.EnsureSuccessStatusCode();
+        }
+
+        public void InitializeRequestHeaders()
+        {
+            _hostName = _environment.GetEnvironmentVariable(EnvironmentSettingNames.AzureWebsiteHostName);
+            _stampName = _environment.GetEnvironmentVariable(EnvironmentSettingNames.WebSiteHomeStampName);
+            _process = Process.GetCurrentProcess();
+        }
+
+        public void StartMetricsPublisher()
+        {
+            InitializeRequestHeaders();
+            _processMonitorTimer = new Timer(OnProcessMonitorTimer, null, TimeSpan.Zero, _memorySnapshotInterval);
+            _metricsPublisherTimer = new Timer(OnFunctionMetricsPublishTimer, null, TimeSpan.Zero, _metricPublishInterval);
+
+            _logTraceEvent(LogLevel.Information, string.Format("LinuxContainerMetricsPublisher: Starting metrics publisher for container : {0}", _containerName));
+        }
+
+        private void OnProcessMonitorTimer(object state)
+        {
+            _process.Refresh();
+            var commitSizeBytes = _process.WorkingSet64;
```

Comment:
Need to run this by @paulbatum to make sure this is the right value to use.

Reply:
We had checked this with the cosine team as well.